public/app/features/templating/template_srv.ts

```diff
@@ -347,6 +349,55 @@ export class TemplateSrv implements BaseTemplateSrv {
     });
   }
 
+  getVariablesMapInTemplate(
+    target: string,
+    scopedVars: ScopedVars,
+    format?: string | Function
+  ): Record<string, VarValue> {
+    const values: Record<string, VarValue> = {};
+
+    this.replaceInVariableRegex(target, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
+      const variableName = var1 || var2 || var3;
+      const variableDisplayName =
+        var1 || var2 || (var3 !== undefined && fieldPath !== undefined) ? `${var3}.${fieldPath}` : var3;
+      const fmt = fmt2 || fmt3 || format;
+      const value = this.getVariableValue(variableName, fieldPath, scopedVars);
+      if (value !== null && value !== undefined) {
+        const variable = this.getVariableAtIndex(variableName);
+        const text = this.getVariableText(variableName, value, scopedVars);
+        values[variableDisplayName] = this.formatValue(value, fmt, variable, text);
+      } else {
+        values[variableDisplayName] = undefined;
+      }
+
+      // Don't care about the result anyway
+      return '';
+    });
+
+    return values;
+  }
+
+  /* 
+    The replace function, for every match, will return a function that has the full match as a param
+    followed by one param per capture group of the variable regex.
+
+    See the definition of this.regex for further comments on the variable definitions.
+  */
+  private replaceInVariableRegex(
```

Comment:
It looks a bit off in my IDE:

<img width="943" alt="Screenshot 2023-03-13 at 14 02 44" src="https://user-images.githubusercontent.com/745532/224709560-0424796a-83eb-4dd7-bcda-5d20acae316f.png">

Something like the below would work better but not sure if in all IDEs:

```typescript
  /**
   * The replace function, for every match, will return a function that has the full match as a param
   * followed by one param per capture group of the variable regex.
   *
   * See the definition of this.regex for further comments on the variable definitions.
   */
  private replaceInVariableRegex(
    text: string,
    replace: (
      fullMatch: string, //     $simpleVarName   [[squareVarName:squareFormat]]   ${curlyVarName.curlyPath:curlyFormat}
      simpleVarName: string, // simpleVarName                  -                                     -
      squareVarName: string, //        -                squareVarName                                -
      squareFormat: string, //         -                squareFormat                                 -
      curlyVarName: string, //         -                      -                                curlyVarName
      curlyPath: string, //            -                      -                                  curlyPath
      curlyFormat: string //           -                      -                                 curlyFormat
    ) => string
  ) {
    return text.replace(this.regex, replace);
  }
```

<img width="993" alt="Screenshot 2023-03-13 at 14 03 09" src="https://user-images.githubusercontent.com/745532/224709707-f595f408-86b6-4590-a522-e653a01868a5.png">

Reply:
Huh, sorry about that. Fixed!